app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore_animal: Herbivore) -> None:
+        if (not isinstance(herbivore_animal, Carnivore)
+                and not herbivore_animal.hidden):
+            herbivore_animal.health -= 50
+            if herbivore_animal.health <= 0:
+                for animal in Animal.alive:
+                    if animal.name == herbivore_animal.name:
+                        Animal.alive.remove(animal)
```

Comment:
This is overcomplicated. `Animal.alive` contains is a  list of Animal instances. On line 30 you check if `herbivore_animal` is dead, so what would be the next step?